### DISCUSSION TITLE
Concerns Regarding Privacy and Accuracy in Current Approach

### DISCUSSION THREAD

**Discussion by Lakr233** [wavetermdev/waveterm#90] [General]
I hope you doing well.

Firstly, if you're considering to keep the current design, it's essential to fork a copy of the user shell binary source code. This approach allows you to add new features while ensuring that the environment is handled correctly. Neglecting this step can lead to significant issues and an unsatisfactory user experience. **In my current configuration, your product fails to correctly identify the binary's location, consequently not executing and accurately retrieving the specifics of the $PATH variable.**

Secondly, I'd like to address the practice of automatically installing mshell on remote hosts. It's important to recognize that such actions can be perceived as intrusive and potentially harmful by users. This kind of automatic installation can give users an uncomfortable feeling, akin to a violation of their digital space. It might even raise concerns about unauthorized backdoor installations. Transparency is the key. **Users should be fully aware of and consent to any changes made to their systems.**

Warm regards.

**Comment by sawka** [comment]
Thanks for posting.  I'll address the first concern in this comment (and the 2nd concern in the next comment).

I've been thinking about this -- if we can distribute the shells.  MacOS is stuck with bash v3 and it would be much easier to integrate with bash v5.  The problem is that bash is GPLed and we are Apache so it isn't possible for us to distribute the bash shell directly.  I'm thinking there could be some sort of optional install though, with code hosted in a separate GPL repo to help get around this issue.  When we support zsh though, we should be able to distribute that directly.

Your bug around the shell PATH is a good one.  We can look into adding a configuration setting for "shell path" to help with this.

**Comment by sawka** [comment]
I understand your concern around installing mshell, and we definitely want to be *transparent* around what's going on (good word choice).  Maybe we can address this with a popup the first time mshell is installed on a remote machine (with a confirm/cancel action) + a checkbox to "not show this warning again".  Or maybe we can make that more clear with some extra help text in the "create connection" flow?  What type of flow do you think would make you feel more comfortable?

**Reply by Lakr233** [reply]
It's a thing from "tell user we are going to install" to "do you wish to install?". :)
